src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,129 @@
+        :return: A string containing only the digits from the input value.
+        """
+        return re.sub(r"[^\d+]", "", value)
+
+    def isInvalid(self, value: str) -> t.Optional[str]:
+        """
+        Checks if the provided phone number is valid or not.
+
+        :param value: The phone number to be validated.
+        :return: An error message if the phone number is invalid or None if it is valid.
+
+        The method checks if the provided phone number is valid according to the following criteria:
+        1. The phone number must not be empty.
+        2. The phone number must match the provided or default phone number format.
+        3. The phone number cannot exceed 15 digits, or the specified maximum length if provided (digits only).
+        """
+        if not value:
+            return "No value entered"
+
+        if not self.test.match(value):
+            return "Invalid phone number entered"
+
+        # make sure max_length is not exceeded.
+        is_invalid: t.Optional[str] = super().isInvalid(self._extract_digits(value))
+
+        if is_invalid:
+            return is_invalid
+        return None
+
+    def singleValueFromClient(
+        self, value: str, skel: t.Any, bone_name: str, client_data: t.Any
+    ) -> t.Tuple[t.Optional[str], t.Optional[t.List[ReadFromClientError]]]:
+        """
+        Processes a single value from the client, applying the default country code if necessary and validating the
+        phone number.
+
+        :param value: The phone number provided by the client.
+        :param skel: Skeleton data (not used in this method).
+        :param bone_name: The name of the bone (not used in this method).
+        :param client_data: Additional client data (not used in this method).
+        :return: A tuple containing the processed phone number and an optional list of errors.
+        """
+        value = value.strip()
+
+        # Replace country code starting with 00 with +
+        if value.startswith("00"):
+            value = "+" + value[2:]
+
+        # Apply default country code if none is provided and default_country_code is set
+        if self.default_country_code and value[0] != "+":
+            if value.startswith("0"):
+                value = value[1:]  # Remove leading 0 from city code
+            value = f"{self.default_country_code} {value}"
+
+        err: t.Optional[str] = self.isInvalid(value)
+        if err:
+            return self.getEmptyValue(), [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, err)]
+
+        return value, None
+
+    def structure(self) -> t.Dict[str, t.Any]:
+        """
+        Returns the structure of the PhoneBone, including the test regex pattern.
+
+        :return: A dictionary representing the structure of the PhoneBone.
+        """
+        return super().structure() | {
+            "test": self.test.pattern if self.test else "",
+            "countries": self.countries,
```

Comment:
Hello @KadirBalku, thanks for all improvements on this issue. I'm unhappy with this part of the pull request.

1. The JSON-file `country_information.json` is read into memory from the PhoneBone
2. The content is then assigned to `self.countries` of the PhoneBone
3. `self.countries`, the entire JSON-content from `country_information.json` is then passed to the client. 

This reveals to me that `country_information.json` does only contain information which is necessary to the client, and does nothing in the logic of the PhoneBone. Therefore, I would like to propose to move the content of `country_information.json` to the PhoneBone client source code.